From edcd1f6a875cbbde5803662c595a7485da14d391 Mon Sep 17 00:00:00 2001
Message-Id: <edcd1f6a875cbbde5803662c595a7485da14d391.1684999160.git.stefan@agner.ch>
In-Reply-To: <c2dc2a9940d1f9809403d4dfb28e1f942ef90bab.1684999160.git.stefan@agner.ch>
References: <c2dc2a9940d1f9809403d4dfb28e1f942ef90bab.1684999160.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Tue, 17 Jan 2023 19:31:59 +0100
Subject: [PATCH] [rest] remove superfluous space in HTTP status line (#1713)

A space is already added in Response::Serialize(). According to the HTTP
specification there should only be a single space between the protcol
version and the status code.

(cherry picked from commit 10a181ee914eb9641eaf416e3747a0a360a41ede)
---
 src/rest/response.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rest/response.cpp b/src/rest/response.cpp
index 2cdfc80ea5..bcf63a6c31 100644
--- a/src/rest/response.cpp
+++ b/src/rest/response.cpp
@@ -45,7 +45,7 @@ Response::Response(void)
     , mComplete(false)
 {
     // HTTP protocol
-    mProtocol = "HTTP/1.1 ";
+    mProtocol = "HTTP/1.1";
 
     // Pre-defined headers
     mHeaders["Content-Type"]                 = OT_REST_RESPONSE_CONTENT_TYPE_JSON;
-- 
2.40.1

